@xen-orchestra/xva/_writeDisk.mjs

```diff
@@ -21,9 +23,17 @@ export default async function addDisk(pack, vhd, basePath) {
     const data = await readChunkStrict(stream, Math.min(chunk_length, remaining))
     lastBlockLength = data.length
     remaining -= lastBlockLength
-
-    if (counter === 0 || !data.equals(empty)) {
+    if (
+      // write first block
+      counter === 0 ||
+      // write all non empty blocks
+      !data.equals(empty) ||
+      // write one block from time to time to ensure there is no timeout
+      // occurring while passing empty blocks
+      Date.now() - lastBlockWrittenAt > MAX_INTERVAL_BETWEEN_BLOCKS
+    ) {
       written = true
+      lastBlockWrittenAt = Date.now()
```

Comment:
If the write is a bit slow, would it be best to update this var after the write instead of before?